### PR TITLE
[WIP]Custom filenames

### PR DESCRIPTION
**Pull request by bwklein**
This is an initial effort to add in the ability to accept input from a user for text added into a custom filename for images created by the package.

### CODE REVIEW

File: lib/custom-filename-view.coffee [hunk 1]
```diff
@@ -0,0 +1,52 @@
+{View, TextEditorView} = require 'atom-space-pen-views'
+
+class CustomFilenameView extends View
+  @content: ->
+  @div class: 'custom-filename', =>
+    @subview 'filenameText', new TextEditorView(mini: true)
```
Comment (bwklein) [inline lib/custom-filename-view.coffee]: I don't know how to refer to this mini editor in the code below when trying to get or set the text entered by the user.

File: lib/custom-filename-view.coffee [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    # Set the customFileString variable to the user input.
+    return customFileString = "-#{userInput}"
+
+  open: ->
+    #console.log 'Opening Input View'
+    return if @modalPanel.isVisible()
+    #console.log 'Input View is not Visible'
+    @modalPanel.show()
+    #console.log 'Show Input View'
+    @modalPanel.filenameText.focus()
```
Comment (bwklein) [inline lib/custom-filename-view.coffee]: Syntax to bring focus to the filenameText element?

File: lib/custom-filename-view.coffee [hunk 3]
```diff
@@ -0,0 +1,52 @@
+{View, TextEditorView} = require 'atom-space-pen-views'
+
+class CustomFilenameView extends View
+  @content: ->
+  @div class: 'custom-filename', =>
```
Comment (ldez) [inline lib/custom-filename-view.coffee]: you have an indent problem here

**Comment by ldez** [comment]
Sorry for now but it's 3 am for me, I need some sleep but tomorrow I would do a code review.

**Comment by bwklein** [comment]
@ldez do you know of a better way to prompt the user for an input string and pull it into a variable within Atom?

File: lib/custom-filename-view.coffee (hunk 2 above)
Comment (ldez) [inline lib/custom-filename-view.coffee]: I think:
```coffee
  @modalPanel.focus()
  @modalPanel.getModel().scrollToCursorPosition()
```

**Comment by ldez** [comment]
You way it's the good way I think :+1:

**Pull request closed without merging**